pkg/test/resourcefixture/testdata/basic/spanner/v1beta1/spanneinstance-unset-compute-fields/_http.log

```diff
@@ -0,0 +1,339 @@
+    "endTime": "2024-04-01T12:34:56.123456Z",
+    "expectedFulfillmentPeriod": "FULFILLMENT_PERIOD_NORMAL",
+    "instance": {
+      "config": "projects/${projectId}/instanceConfigs/regional-us-west1",
+      "displayName": "Spanner Instance Sample",
+      "labels": {
+        "cnrm-test": "true",
+        "label-one": "value-one",
+        "managed-by-cnrm": "true"
+      },
+      "name": "projects/${projectId}/instances/spannerinstance-sample-${uniqueId}",
+      "nodeCount": 2,
+      "processingUnits": 2000,
+      "state": "READY",
+      "updateTime": "2024-04-01T12:34:56.123456Z"
+    },
+    "startTime": "2024-04-01T12:34:56.123456Z"
+  },
+  "name": "projects/${projectId}/instances/spannerinstance-sample-${uniqueId}/operations/${operationID}",
+  "response": {
+    "@type": "type.googleapis.com/google.spanner.admin.instance.v1.Instance",
+    "config": "projects/${projectId}/instanceConfigs/regional-us-west1",
+    "createTime": "2024-04-01T12:34:56.123456Z",
+    "displayName": "Spanner Instance Sample",
+    "labels": {
+      "cnrm-test": "true",
+      "label-one": "value-one",
+      "managed-by-cnrm": "true"
+    },
+    "name": "projects/${projectId}/instances/spannerinstance-sample-${uniqueId}",
+    "nodeCount": 2,
+    "processingUnits": 2000,
+    "state": "READY",
+    "updateTime": "2024-04-01T12:34:56.123456Z"
+  }
+}
+
+---
+
+GET https://spanner.googleapis.com/v1/projects/${projectId}/instances/spannerinstance-sample-${uniqueId}?alt=json
+Content-Type: application/json
+User-Agent: Terraform/ (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google-beta/kcc/controller-manager
+
+200 OK
+Cache-Control: private
+Content-Type: application/json; charset=UTF-8
+Server: ESF
+Vary: Origin
+Vary: X-Origin
+Vary: Referer
+X-Frame-Options: SAMEORIGIN
+X-Xss-Protection: 0
+
+{
+  "config": "projects/${projectId}/instanceConfigs/regional-us-west1",
+  "createTime": "2024-04-01T12:34:56.123456Z",
+  "displayName": "Spanner Instance Sample",
+  "labels": {
+    "cnrm-test": "true",
+    "label-one": "value-one",
+    "managed-by-cnrm": "true"
+  },
+  "name": "projects/${projectId}/instances/spannerinstance-sample-${uniqueId}",
+  "nodeCount": 2,
+  "processingUnits": 2000,
+  "state": "READY",
+  "updateTime": "2024-04-01T12:34:56.123456Z"
+}
+
+---
+
+PATCH https://spanner.googleapis.com/v1/projects/${projectId}/instances/spannerinstance-sample-${uniqueId}?alt=json
+Content-Type: application/json
+User-Agent: Terraform/ (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google-beta/kcc/controller-manager
+
+{
+  "fieldMask": "displayName",
+  "instance": {
+    "displayName": "New spanner Instance Sample",
+    "labels": {
+      "cnrm-test": "true",
+      "label-one": "value-one",
+      "managed-by-cnrm": "true"
+    },
+    "name": "projects/${projectId}/instances/%!s(\u003cnil\u003e)",
+    "nodeCount": 2,
+    "processingUnits": 2000
```

Comment:
I see the TF-based considers unset as a no-op. I think that is not right (unless the GCP service assigns a default value in creation). If no defaulting, when migrating to the direct approach, we want to change this behavior to **require** user to configure one of the field (unless autoscaling is turned on).

Reply:
if there is no value at creation, TF assign 1 as a value for nodecount. I agree with the check to ask for at least 1 fields, which would keep the controller and API in sync.

Reply:
I can update the test case in the direct controller PR.